src/Pages/3DPrintingConsole/3DConsole.js

```diff
@@ -58,8 +47,8 @@ export default class PrintConsole3D extends React.Component {
   parameter: Json object of object to be deleted
   Search for object in db using name and color then delete
   */
-  async deleteData (jsonObject) {
-    await delete3DPrintRequest(jsonObject, this.state.authToken)
+  handleDeleteData = async requestToDelete => {
+    await delete3DPrintRequest(requestToDelete, this.state.authToken)
     this.callDatabase() // reload database
```

Comment:
If we are on a slow connection, calling the database every time we update or delete a request slows things down a bunch (imagine we have +100 items to render).

We can fix this by using [`array.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to simply reflect the changes after a deletion.

so something like this
```js
await delete3DPrintRequest(requestToDelete, this.state.authToken)
this.setState({
  data: this.state.data.filter(request => !request.email.includes(request._id))
})
```